src/Features/CSharp/Portable/StringIndentation/CSharpStringIndentationService.cs

```diff
@@ -19,69 +20,62 @@
 namespace Microsoft.CodeAnalysis.CSharp.StringIndentation;
 
 [ExportLanguageService(typeof(IStringIndentationService), LanguageNames.CSharp), Shared]
-internal sealed class CSharpStringIndentationService : IStringIndentationService
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class CSharpStringIndentationService() : IStringIndentationService
 {
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public CSharpStringIndentationService()
-    {
-    }
-
     public async Task<ImmutableArray<StringIndentationRegion>> GetStringIndentationRegionsAsync(
         Document document, TextSpan textSpan, CancellationToken cancellationToken)
     {
         var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
         var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-        using var _ = ArrayBuilder<StringIndentationRegion>.GetInstance(out var result);
-
-        Recurse(text, root, textSpan, result, cancellationToken);
-
-        return result.ToImmutableAndClear();
-    }
-
-    private static void Recurse(
-        SourceText text,
-        SyntaxNode root,
-        TextSpan textSpan,
-        ArrayBuilder<StringIndentationRegion> result,
-        CancellationToken cancellationToken)
-    {
-        cancellationToken.ThrowIfCancellationRequested();
+        using var result = TemporaryArray<StringIndentationRegion>.Empty;
 
         using var _ = ArrayBuilder<SyntaxNode>.GetInstance(out var nodeStack);
         nodeStack.Add(root);
 
         while (nodeStack.TryPop(out var node))
         {
-            // DoNot' bother recursing into nodes that don't hit the requested span, they can never contribute 
-            // regions of interest.
-            if (!node.Span.IntersectsWith(textSpan))
```

Comment:
>if (!node.Span.IntersectsWith(textSpan)) [](http://example.com/codeflow?start=12&length=40)

seems like we did this inefficiently in a lot of places.

Reply:
Yup!  Tackling the ones that show up in traces first